src/components/Topnav.tsx

```diff
@@ -1,8 +1,66 @@
-import React from "react";
+import { Avatar, Navbar } from "flowbite-react";
+import LogoSmall from "../assets/LogoWhiteSmall.svg";
+import StudentAvatar from "../assets/StudentAvatar.svg";
+import ProfessorAvatar from "../assets/ProfessorAvatar.svg";
+import { useState } from "react";
+import { CATEGORIES } from "../utils/TopnavConstants";
+import { useLocation } from "react-router";
+import { Link } from "react-router-dom";
 
-// TODO
 function Topnav() {
-  return <div></div>;
+  const { search, pathname } = useLocation();
+  const [toggle, setToggle] = useState(false);
+  const queryParams = new URLSearchParams(search);
+  const targetCategory = queryParams.get("category") || CATEGORIES[0];
+
+  function isTargetCategory(category: string) {
+    return category === targetCategory;
+  }
+
+  return (
+    <div className="text-white [&>*]:bg-blue-500">
```

Comment:
"[&>*]:.." affects all direct children and "bg-blue-500" is pretty similar to the color from Figma mocks